dhis2/dhis2-core
# feat: limits for category combinations [DHIS2-18521]

### Summary
Adds 3 configurations for new limits around categories that are validated on create and update.

* `metadata.categories.max_options`: The maximum number of category options in a single category
* `metadata.categories.max_per_combo`: The maximum number of categories per category combo
* `metadata.categories.max_combinations`: The maximum for the product of the number of options in a category combo

### Automatic Testing
New controller tests added for each of the validations.

### Manual Testing
Test limit max categories in a combo:

* goto maintenance app
* create a new category combo
* enter a name, select 6 categories and press save
* check a error message is shown that explains that a combo can only have 5 categories

Test limit max options per category:

* goto maintenance app
* edit an existing category
* select more than 31 options (move to right) and press save
* check a error message is shown that explains that a category can only have 31 options or less

Test limit max combinations:

* goto maintenance app
* find or create 3 categories with each have at least 8 (or more) options
* create a category combo with these 3 categories and press save
* check a error message is shown that explains that a category combo can only have more than 500 combinations
